### PR TITLE
RRTMGP coupling to Thompson MP in ccpp-physics

### PR DESCRIPTION
**Pull request by dustinswales**
## Description
This PR contains an accumulation of changes to the coupling of the RRTMGP radiation and the cloud-microphysics.
-) New coupling of RRTMGP to Thompson MP
-) Include scattering in LW all-sky rte calculation (doGP_lwscat = .true.)
-) Use of internal RRTMGP cloud-optics (doGP_cldoptics_LUT or doGP_cldoptics_PADE = .true).
-) Adjust surface upwelling LW surface fluxes between RRTMGP calls using the Jacobian of the upwelling LW flux (use_LW_jacobian = .true.). F_adj = F_o + dT*Jacobian
-) Bug fix in RRTMGP interstitial data

## Testing
Answers to the v16beta baselines using RRTMGP need to be updated. The bug fix in the interstitial data greatly reduces differences between RRTMGP and RRTMG forecasts. Baselines need to be created for the new SDF.

These changes were tested using Intel on Hera. 

## Dependencies
https://github.com/NCAR/ccpp-physics/pull/528

### CODE REVIEW

**Comment by climbfuji** [comment]
@dustinswales please update the title of this PR when you get a chance - thank you!

File: gfsphysics/GFS_layer/GFS_typedefs.F90 [hunk 1]
```diff
@@ -7065,8 +7075,9 @@ subroutine interstitial_rad_reset (Interstitial, Model)
         Interstitial%cwm       = clear_val
       end if
     end if
-
+
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.F90]: Don't need these whitespaces

File: gfsphysics/GFS_layer/GFS_typedefs.F90 [hunk 2]
```diff
@@ -7288,7 +7299,7 @@ subroutine interstitial_phys_reset (Interstitial, Model)
     Interstitial%tseal           = clear_val
     Interstitial%tsfc_ice        = huge
     Interstitial%tsfc_land       = huge
-    Interstitial%tsfc_ocean      = huge
+    Interstitial%tsfc_ocean      = huge
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.F90]: Don't need these whitespaces

File: gfsphysics/GFS_layer/GFS_typedefs.F90 [hunk 3]
```diff
@@ -7676,7 +7687,7 @@ subroutine interstitial_print(Interstitial, Model, mpirank, omprank, blkno)
     write (0,*) 'sum(Interstitial%tsfa            ) = ', sum(Interstitial%tsfa            )
     write (0,*) 'sum(Interstitial%tsfc_ice        ) = ', sum(Interstitial%tsfc_ice        )
     write (0,*) 'sum(Interstitial%tsfc_land       ) = ', sum(Interstitial%tsfc_land       )
-    write (0,*) 'sum(Interstitial%tsfc_ocean      ) = ', sum(Interstitial%tsfc_ocean      )
+    write (0,*) 'sum(Interstitial%tsfc_ocean      ) = ', sum(Interstitial%tsfc_ocean      )
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.F90]: Don't need these whitespaces

**Review by climbfuji — CHANGES_REQUESTED** [review]

**Comment by climbfuji** [comment]
@dustinswales I hate to bug you, but this PR is still pointing to hash ea5e44fcc691c9a5fedcb96f13e0ca85259c7844 of ccpp-physics, and not 37313e512087fc47f51bf12375e724c80fa3c18c. Can you fix that, please, and then update fv3atm in ufs-weather-model (I know, what a hassle every time).

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by climbfuji — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**